ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java

```diff
@@ -223,7 +218,7 @@ private void updateTheAccounts(
       final BonsaiAccount updatedAccount = bonsaiValue.getUpdated();
       try {
         if (updatedAccount == null) {
-          final Hash addressHash = Hash.hash(accountKey);
+          final Hash addressHash = preImageProxy.hashAndSavePreImage(accountKey);
```

Comment:
it's clearly too much modification just for testing. I will be happy if we have a way to avoid that

Reply:
I will look at deferring to a hash() method we can call in BonsaiWorldState that we can override in the reference test worldstate with a preimageProxy.  Before I do I want to collect some cache hit metrics and perf numbers on mainnet with an in-memory preImage proxy to see if using the preimage proxy would be beneficial vs hashing every time.

Reply:
In BonsaiWorldState and the accumulator I added a hashAndSavePreImage function that by default does not save pre-images and just override this in the referencetest worldstate.  The only tradeoff here is that I had to make the accumulator mutable via a protected method.